src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -2250,6 +2250,42 @@ mgp_error mgp_graph_delete_edge(struct mgp_graph *graph, mgp_edge *edge) {
   });
 }
 
+#ifdef MG_ENTERPRISE
+namespace {
+void NextPermitted(mgp_vertices_iterator *it) {
```

Comment:
```suggestion
void NextPermitted(mgp_vertices_iterator &it) {
```

On top of that I don't see why should it be protected with the `MG_ENTERPRISE` macro. The regarding field (`it->graph->ctx->fine_grained_access_checker`) is always there, even in community versions. I think this extra function call and `nullptr` check is a reasonable price to pay for getting rid of some `#ifdef`s.